_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -730,6 +734,8 @@ mfxStatus MFX_CDECL VP9DECODERoutine(void *p_state, void * /* pp_param */, mfxU3
     if (data.currFrameId != -1)
        decoder.m_FrameAllocator->DecreaseReference(data.currFrameId);
 
+    decoder.CheckFreeBufferedSurface(data.currFrameId);
```

Comment:
The same as upper - I think it's better to move this to "if (data.currFrameId != -1)" condition on line 734